web-ui/main.ml

```diff
@@ -1,101 +1,137 @@
 open Lwt.Infix
-open Astring
+module Router = struct
 
-let setup_log default_level =
-  Prometheus_unix.Logging.init ?default_level ()
-
-module Server = Cohttp_lwt_unix.Server
-
-let normal_response x =
-  x >|= fun x -> `Response x
-
-let headers content_type max_age =
-  Cohttp.Header.of_list [
-    ("Content-Type", content_type);
-    ("Cache-Control", Printf.sprintf "public, max-age=%d;" max_age);
-  ]
+  (* ocaml-crunch is used to generate the module Static.
+   See also https://github.com/aantron/dream/tree/master/example/w-one-binary *)
+  let loader root path _request =
+    match Static.read (Filename.concat root path) with
+    | None -> Dream.empty `Not_Found
+    | Some asset -> Dream.respond asset
 
-let static ~content_type ?(max_age=86400) body =
-  let headers = headers content_type max_age in
-  Server.respond_string ~status:`OK ~headers ~body ()
+  let t ci = Dream.router
```

Comment:
Shouldn't this module live as a standalone file? IMHO, it feels like it will simplify the `main` module and keep the idea that it's the "late level" where we put pieces together.

Reply:
I thought about this when I extracted the module and didn't actually do anything about it at the time. Thanks for the suggestion 👍🏾